tests/Unit/AutoRouteManagerTest.php

```diff
@@ -11,154 +11,360 @@
+use Symfony\Cmf\Component\RoutingAuto\UriContextCollection;
+use Symfony\Cmf\Component\RoutingAuto\UriContextCollectionBuilder;
+use Symfony\Cmf\Component\RoutingAuto\UriGeneratorInterface;
 
+/**
+ * @testdox The manager
+ */
 class AutoRouteManagerTest extends \PHPUnit_Framework_TestCase
 {
     private $adapter;
     private $uriGenerator;
     private $defunctRouteHandler;
     private $collectionBuilder;
+    private $manager;
 
     public function setUp()
     {
-        $this->adapter = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\AdapterInterface');
-        $this->uriGenerator = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\UriGeneratorInterface');
-        $this->defunctRouteHandler = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\DefunctRouteHandlerInterface');
-        $this->collectionBuilder = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\UriContextCollectionBuilder');
+        $this->adapter = $this->prophesize(AdapterInterface::class);
+        $this->uriGenerator = $this->prophesize(UriGeneratorInterface::class);
+        $this->defunctRouteHandler = $this->prophesize(DefunctRouteHandlerInterface::class);
+        $this->collectionBuilder = $this->prophesize(UriContextCollectionBuilder::class);
 
         $this->manager = new AutoRouteManager(
             $this->adapter->reveal(),
             $this->uriGenerator->reveal(),
             $this->defunctRouteHandler->reveal(),
             $this->collectionBuilder->reveal()
         );
-
-        $this->collection = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\UriContextCollection');
-        $this->context1 = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\UriContext');
-        $this->context2 = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\UriContext');
-        $this->autoRoute1 = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\Model\AutoRouteInterface');
-        $this->autoRoute2 = $this->prophesize('Symfony\Cmf\Component\RoutingAuto\Model\AutoRouteInterface');
-
-        $this->subject = new \stdClass();
     }
 
     /**
-     * It should delegate the generation of the collection and resolve the URIs
-     * in it (for non-existing URIs).
-     * It should handle defunct routes.
+     * Provides the routes configuration for each tested use case.
+     *
+     * Each dataset is an array of routes configuration. Each configuration
+     * is an associative array containing:
+     *  - generatedUri (string): the URI made by the URI generator for this route,
+     *  - locale (string): the locale
+     *  - existsInDatabase (boolean): is there an existing persisted route matching the same URI,
+     *  - expectedUri (string): the URI expected to be set on the current route.
+     *
+     * If the generated URI and the expected one are different, it means that
+     * the conflict resolver should be called by the manager.
      */
-    public function testBuildCollection()
+    public function routesConfigurations()
```

Comment:
OK. Usually, I rely on the annotations. I do the same with the test methods.
I will modify it in order to follow your convention.